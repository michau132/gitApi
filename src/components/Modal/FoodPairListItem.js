import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const FoodPairListItemStyle = styled.li`
  padding-left: 15px;
  position: relative;
  
  :before {
    content: '';
    position: absolute;
    top: 18px;
    left: 5px;
    height: 2px;
    width: 5px;
    background: ${({ theme }) => theme.color.modal}
  }
`;

const FoodPairListItem = ({ text }) => (
  <FoodPairListItemStyle>
    {text}
  </FoodPairListItemStyle>
);

FoodPairListItem.propTypes = {
  text: PropTypes.string.isRequired,
};

export default FoodPairListItem;
