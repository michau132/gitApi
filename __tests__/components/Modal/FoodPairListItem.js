import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FoodPairListItem from '../../../src/components/Modal/FoodPairListItem';

describe('<FoodPairListItem /> component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      text: 'meat',
    };
    wrapper = shallow(<FoodPairListItem {...props} />);
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
