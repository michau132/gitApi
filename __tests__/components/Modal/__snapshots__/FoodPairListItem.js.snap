// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FoodPairListItem /> component matching snapshot 1`] = `
<styled.li>
  meat
</styled.li>
`;
